Library/Formula/luvi.rb

```diff
@@ -0,0 +1,19 @@
+class Luvi < Formula
+  desc "A project in-between luv and luvit, lua packages."
+  homepage "https://github.com/luvit/luvi"
+  url "https://github.com/luvit/luvi/releases/download/v2.1.8/luvi-src-v2.1.8.tar.gz"
+  sha256 "1ef449c6a76bcf7a03c8bbcdd5423b0366d63b10c52fd463ca2ba8aaf9248781"
+
+  depends_on "cmake" => :build
+
+  def install
+    system "make", "regular-asm"
+    system "make"
+    bin.install "build/luvi"
+  end
+
+  test do
+    system "#{bin}/luvi", "samples/test.app", "--", "1", "2", "3", "4"
```

Comment:
`samples/test.app` won't exist at this point. Buildpath and Testpath are separate entities that are never in the same place at once. You'll either need to write the various necessary files in as something like:

```ruby
test do
  (testpath/"testing.test").write <<-EOS.undent
    something to test
  EOS

  system bin/"luvi", "testing.test", "something", "another flag", "so on"
end
```

Or you can do something like this:

```ruby
def install
  system "make", "regular-asm"
  system "make"
  bin.install "build/luvi"
  (pkgshare/"examples").install Dir["samples/*"]
end
```

And then:

```ruby
test do
  system bin/"luvi", pkgshare/"examples/test.app", "--", "1", "2", "3", "4"
end
```

You may find the second way nicer/easier.